ts/sagas/authentication.ts

```diff
@@ -73,6 +89,71 @@ export function* loadSession(): IterableIterator<Effect> {
   }
 }
 
+/**
+ * Listen to APP_STATE_CHANGE_ACTION and if needed force the user to insert the PIN
+ */
+// tslint:disable-next-line:cognitive-complexity
+export function* watchApplicationActivity(): IterableIterator<Effect> {
+  // tslint:disable-next-line:no-let
+  let lastState: ApplicationState = "active";
+  // tslint:disable-next-line:no-let
+  let lastUpdateAt = -1;
+
+  // We will use this to save and then restore the navigation
+  // tslint:disable-next-line:no-let
+  let navigationState: NavigationState = INITIAL_STATE;
+
+  while (true) {
+    const action: ApplicationStateAction = yield take(APP_STATE_CHANGE_ACTION);
+
+    const newState: ApplicationState = action.payload;
+    const newUpdateAt = new Date().getTime();
+
+    const timeElapsed = newUpdateAt - lastUpdateAt;
+    if (lastState === "active" && newState === "background") {
+      // Save the navigation state so we can restore in case the PIN login is needed
+      // tslint:disable-next-line:saga-yield-return-type
+      navigationState = yield select(navigationStateSelector);
+
+      // Push the BackgroundScreen
+      yield put(
+        NavigationActions.navigate({
+          routeName: ROUTES.BACKGROUND
+        })
+      );
+    } else if (
+      lastState === "background" && // The app was in background
+      newState === "active" // The app is now active
+    ) {
+      if (timeElapsed > backgroundActivityTimeout * 1000) {
```

Comment:
it seems like you can move the computation of `timeElapsed` here since it's used only inside this `else` block